### ISSUE TITLE
Add openPMD output for particles

### ISSUE THREAD

**Issue by RemiLehe** [ECP-WarpX/WarpX#427]
It would be great to have output for the particles, similarly to the openPMD output that was implemented for the fields (see #80 #256)

**Comment by RemiLehe** [comment]
Some more info on this:

- Here is how we currently dump the particles into the AMReX file format ("plotfile"): 
https://github.com/ECP-WarpX/WarpX/blob/master/Source/Diagnostics/ParticleIO.cpp#L77
We would like to keep this, but add more code to dump the same information, using the openPMD api.

- In case this is useful, the structure of particle container (in AMREx/WarpX) is described here:
https://amrex-codes.github.io/amrex/docs_html/Particle.html#the-particlecontainer

- Also, here is how we dump the *fields* (not the particles) using the openPMD-api:
https://github.com/ECP-WarpX/WarpX/blob/master/Source/Diagnostics/FieldIO.cpp#L115

- Instructions for building the code with openPMD-api are here:
https://warpx.readthedocs.io/en/latest/building/openpmd.html
The backend (`hdf5` or `adios`) can be chosen by with this input parameter:
https://warpx.readthedocs.io/en/latest/running_cpp/parameters.html#diagnostics-and-output

- This could be tested using this input script:
https://github.com/ECP-WarpX/WarpX/blob/master/Examples/Tests/Langmuir/inputs.multi.rt
In other words, after building WarpX with openPMD support, the capability can be tested by doing:
```
./warpx.exe input.multi.rt
```

**Comment by ax3l** [comment]
For particles, it might be necessary to prepare the AMReX particle data, sometimes a struct of array, into per-rank and per-quantity (openPMD speach: *record-component*) contiguous arrays.

**Comment by RemiLehe** [comment]
Yes, note that we can convert the positions from the array-of-struct format to a set of arrays, using `WarpXParIter::GetPosition`: https://github.com/ECP-WarpX/WarpX/blob/master/Source/Particles/WarpXParticleContainer.cpp#L28

**Comment by ax3l** [comment]
cc-ing @guj

**Comment by RemiLehe** [comment]
Closed by #514
